### PR TITLE
:recycle: refactored cash-flow report from victory to recharts

### PR DESCRIPTION
**Pull request by MatissJanis**

### CODE REVIEW

File: packages/desktop-client/e2e/reports.test.js-snapshots/Reports-loads-cash-flow-graph-and-checks-visuals-1-chromium-linux.png
Comment (youngcw) [inline packages/desktop-client/e2e/reports.test.js-snapshots/Reports-loads-cash-flow-graph-and-checks-visuals-1-chromium-linux.png:1]: Would it be possible to get the thin bars again? Or just thinner?

Comment (Teprifer) [reply packages/desktop-client/e2e/reports.test.js-snapshots/Reports-loads-cash-flow-graph-and-checks-visuals-1-chromium-linux.png:1]: I kinda like the fatter bars, they add vibrancy the thin ones always seemed a bit harder to read. No shaming please. :)

**Comment by Teprifer** [comment]
(Images from the same budget file.)

Y axis has less increments, not inherently bad, just looks like it's scaling a bit above where the max value is.

Live: 
image

PR
image

I think it'd be good to have the number formatting applies to the axis though(in my case, commas), per live.


X axis is showing more increments, I'm in favour as it's still not crowded:
Live:
image
PR:
image

Minor:
Privacy mode replaced the Y axis values with "..." instead of bluring, this is inconsistent with how the feature affects values elsewhere. 
- Mouse over doesn't show the value (it does in live with blur).
Worth noting that on live they aren't changed by privacy mode at all (a bug imo).

**Comment by youngcw** [comment]
the x-axis is funky when looking at small date ranges.  The spacing is not consistent.
image

File: packages/desktop-client/e2e/reports.test.js-snapshots/Reports-loads-cash-flow-graph-and-checks-visuals-1-chromium-linux.png
Comment (youngcw) [reply packages/desktop-client/e2e/reports.test.js-snapshots/Reports-loads-cash-flow-graph-and-checks-visuals-1-chromium-linux.png:1]: The wide bars feel more cluttered to me.  I believe white space is part of Actual's design language and the different bars seem to go against that.

**Comment by MatissJanis** [comment]
Thanks for the review @youngcw and @Teprifer !

Couple of updates:
1. Y axis will now have 8 ticks
2. Y axis ticks will be formatted as numbers (i.e. with coma separators or other formats - depending on users prefs)
3. Bar charts now have max size 


Things that aren't done:
- Y axis + fancy privacy blur mode - sadly this won't be possible as recharts is a SVG drawing; it is possible to do blurs with SVG, but because the ticks are so close to the edge - a sharp edge is visible on the blur thus making it look broken; as for consistency: the three dots approach is used for all recharts implementations we have now that support privacy mode (check custom reports);
- inconsistent X axis gaps: this is actually a feature of recharts; it tries to fit as many ticks on the x-axis as possible. If the tick text is too long: it skips adding these ticks. Which is why you're seeing some gaps where a long text would have been. We could override this logic with manual tick definitions, however, this has a very high likelihood of having overlapping tick contents for some resolutions. It's hard to predict when and how it might happen. Which is why I'd recommend sticking with the default recharts logic here.

File: packages/desktop-client/e2e/reports.test.js-snapshots/Reports-loads-cash-flow-graph-and-checks-visuals-1-chromium-linux.png
Comment (Teprifer) [reply packages/desktop-client/e2e/reports.test.js-snapshots/Reports-loads-cash-flow-graph-and-checks-visuals-1-chromium-linux.png:1]: Can see what you're saying, possibly some room in the middle?

For what it's worth sizes are all over the place in live, although there's always generous spacing as you noted.

One month (Skinny):
image

Three months (Skinny):
image

Six months (Thick):
image

One year (Medium): 
image

All time (Skinny): 
image

**Comment by MatissJanis** [comment]
Found one more bug and patched it: transfers value was not showing in the tooltip.

**Comment by youngcw** [comment]
This is a related issue.  I don't know if we care to actual change anything. #2237
the tl;dr is that the report separates out transfers from expenses, but the report dash card doesn't so the numbers show up different.

**Comment by youngcw** [comment]
> Found one more bug and patched it: transfers value was not showing in the tooltip.

Im still not seeing transfers in the hover tooltip.

**Comment by MatissJanis** [comment]
@youngcw I think that's a fair point, but I'd prefer if we kept the scope here minimal: purely a refactor with minimal changes to the UX/functionality. We can circle back afterwards for improvements (for example: I'd really like to add 2x separate Y-axis instead of 1x to better scale the graph.. or perhaps change the scale to `sqrt`)

**Comment by MatissJanis** [comment]
> > Found one more bug and patched it: transfers value was not showing in the tooltip.
> 
> Im still not seeing transfers in the hover tooltip.

You'd only see them if there were any transfers. So for testing this my recommendation is: open demo or your local app and find a time period with transfer (by hovering on the graph and finding a tooltip that has a transfer in it..).

Once you found one: open the preview link here and locate the same exact timeframe. The transfer should show up here too.

**Comment by youngcw** [comment]
> > > Found one more bug and patched it: transfers value was not showing in the tooltip.
> > 
> > 
> > Im still not seeing transfers in the hover tooltip.
> 
> You'd only see them if there were any transfers. So for testing this my recommendation is: open demo or your local app and find a time period with transfer (by hovering on the graph and finding a tooltip that has a transfer in it..).
> 
> Once you found one: open the preview link here and locate the same exact timeframe. The transfer should show up here too.

Ok, I see it now.  Its being smarter than I expected.

**Comment by Teprifer** [comment]
The animations are darned smooth, but the one for line feels like it takes too long to get to its final state because it is slowing down the closer it gets. It completes well after the bars have finished moving. 

Thoughts, have it take 20-30% less time, either by removing the slow down aspect, or spend more time at the faster rate and have the slow down over a shorter time period. Depending on what's possible or feels good etc.


I agree on having two y-axis, the bars are barely nearly invisible when looking at the smaller time frames in my budget. When doing so I think youngcw's comment above about bar thickness would need to be taken in to account as the bars would be much bigger vertically.

**Comment by joel-jeremy** [comment]
We should not blur the amounts in the tooltip when hovering over the graphs IMO since there will be no way for the user to show those values without disabling privacy mode.

**Comment by MatissJanis** [comment]
Changes:

- removed privacy mode from tooltips (makes sense IMO)
- reduced animation duration from 1.5s to 1s

**Review by youngcw — APPROVED** [review]

**Pull request merged**